pulsar-client-cpp/lib/ClientConnection.cc

```diff
@@ -1382,6 +1385,10 @@ void ClientConnection::close() {
     if (tlsSocket_) {
         tlsSocket_->lowest_layer().close();
     }
+
+    if (executor_) {
+        executor_.reset();
```

Comment:
The program doesn't seem to stop unless `executor_` is reset here. However, the `ClientConnection` destructor is not executed, causing a memory leak.

Reply:
Removed this in this PR.